### PR TITLE
Detect unmounted ESP (Fixes: #1405)

### PR DESCRIPTION
**Pull request by superm1**
This is also 1.3.3 material, early comments appreciated.

There is definitely a memory leak going on with the dbus comms.  Can tell from
```g_atomic_ref_count_dec: assertion 'g_atomitc_int_get (arc) > 0``` showing up in the logs but I'm not sure what i'm doing wrong.  If you spot it, appreciated!

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
I think the corruption is happening around how this is being done:
```
	g_variant_builder_init (&builder, G_VARIANT_TYPE_VARDICT);
	input = g_variant_new ("(a{sv})", &builder);
```

**Comment by superm1** [comment]
I sorted out the leak, and verified this works for me if ESP is not defined in `/etc/fstab` and unmounted.

UDisks2 ends up mounting it in `/media/$user/$FS_LABEL`, and fwupd uses it just fine.

File: plugins/uefi/fu-uefi-common.c [hunk 1]
```diff
@@ -351,8 +356,47 @@ fu_uefi_check_esp_path (const gchar *path, GError **error)
...
+	g_autofree gchar *found_esp = NULL;
+	const gchar *obj = NULL;
+
+	output = fu_uefi_get_block_devices (error);
+	if (output == NULL)
+		return FALSE;
+	g_variant_get (output, "(ao)", &iter);
+	while (g_variant_iter_next (iter, "o", &obj)) {
+		gboolean esp = fu_uefi_obj_is_esp (obj);
+		g_debug ("block device %s: %d", obj, esp);
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-common.c]: maybe `is_isp:%d` for our future selves

Comment (superm1) [reply plugins/uefi/fu-uefi-common.c]: fixed

File: plugins/uefi/fu-uefi-udisks.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#define ESP_DISK_TYPE			"c12a7328-f81f-11d2-ba4b-00a0c93ec93b"
+
+gboolean
+fu_uefi_is_udisks_path (const gchar *path)
+{
+	return g_str_has_prefix (path, "/org/freedesktop/UDisks2/");
+}
+
+static GDBusProxy *
+fu_uefi_get_dbus_proxy (const gchar *path, const gchar *interface,
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-udisks.c]: I feel like all the things in theis file should have a `fu_uefi_udisks_` prefix if they're udisks-specific.

File: plugins/uefi/fu-uefi-udisks.c [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				       NULL, error);
+	if (proxy == NULL) {
+		g_prefix_error (error, "failed to find %s: ", UDISKS_DBUS_SERVICE);
+		return NULL;
+	}
+	return g_steal_pointer (&proxy);
+}
+
+GVariant *
+fu_uefi_get_block_devices (GError **error)
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-udisks.c]: Not sure I approve of "exporting" a GVariant -- can't we convert that to something friendlier? e.g. GPtrArray of gchar -- a GVariant is somewhat of an implementation detail, and it's tricky to express how to sink, or unref it.

File: plugins/uefi/fu-uefi-udisks.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+GPtrArray	*fu_uefi_udisks_get_block_devices	(GError **error);
+gboolean	 fu_uefi_udisks_objpath    		(const gchar *path);
+gboolean	 fu_uefi_udisks_objpath_is_esp		(const gchar *obj);
+gchar		*fu_uefi_udisks_objpath_mount		(const gchar *path, GError **error);
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-udisks.h]: can you make this the same indenting as the other .h file pls.

File: plugins/uefi/fu-uefi-udisks.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+	g_variant_get (val, "s", &str);
+	return g_strcmp0 (str, ESP_DISK_TYPE) == 0;
+
+}
+
+gchar *
+fu_uefi_udisks_objpath_mount (const gchar *path, GError **error)
+{
+	g_autoptr(GDBusProxy) proxy = NULL;
+	g_autoptr(GVariant) val = NULL;
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-udisks.c]: A total nit, but normally `g_autoptr()` are last in the C99 list.

File: plugins/uefi/fu-uefi-udisks.c [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+		return FALSE;
+	}
+	val = g_dbus_proxy_get_cached_property (proxy, "Type");
+	if (val == NULL)
+		return FALSE;
+
+	g_variant_get (val, "s", &str);
+	return g_strcmp0 (str, ESP_DISK_TYPE) == 0;
+
+}
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-udisks.c]: random extra newline

File: plugins/uefi/fu-uefi-udisks.c (hunk 3 above)
Comment (superm1) [reply plugins/uefi/fu-uefi-udisks.c]: fixed

File: plugins/uefi/fu-uefi-common.h [hunk 7]
```diff
@@ -72,7 +72,7 @@ gboolean	 fu_uefi_get_framebuffer_size	(guint32	*width,
 						 guint32	*height,
 						 GError		**error);
 gboolean	 fu_uefi_secure_boot_enabled	(void);
-gchar		*fu_uefi_guess_esp_path		(void);
+gchar		*fu_uefi_guess_esp_path		(GError **error);
```
Comment (hughsie) [inline plugins/uefi/fu-uefi-common.h]: indent pls.

**Pull request merged**